statistics/scalar.go

```diff
@@ -175,3 +177,122 @@ func calcFraction4Datums(lower, upper, value *types.Datum) float64 {
 	}
 	return 0.5
 }
+
+const maxNumStep = 10
+
+func enumRangeValues(low, high types.Datum, lowExclude, highExclude bool) []types.Datum {
+	if low.Kind() != high.Kind() {
```

Comment:
When would this happen? like `(1, +inf]`?

Reply:
Yes.